app/models/ability.rb

```diff
@@ -0,0 +1,34 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, Post, public: true
+    can :read, User, public: true
+
+    return unless user.present?
+
+    can :manage, :all
+
+    return unless user.admin?
+
+    can :manage, :all
```

Comment:
- [ ] Great job so far @stephenkati and @0sugo There are just a few changes I will suggest you implement and you'll be on your way to the next milestone :wink: 
- [ ] Great job including `cancancan` in your project and ensuring that the `ability.rb` file was utilized :clap: However, I think that your authorization logic might not have been well put together.
- [ ] From the project requirement below, you can see that you needed to ensure that the `CanCanCan` ability class defines the set of rules for user-post interactions.
- A user can delete a post if it is theirs or if they have an admin role
- A user can delete a comment if it is theirs or if they have an admin role
- Currently, a user can delete a post or a comment even if it's not theirs. I also noticed that the `role` column is `nil` for all the created users which means none of them are `admins` and as a result shouldn't be able to delete other users' posts and comments. So kindly fix this so as to be in accordance with the project requirements :+1: 
- [ ] From the video, you can see that I was signed in as Tyra but I was able to see the delete button on the user Nahnah's Post and comment. I was also able to delete Nahnah's comment but I was unable to delete Nahnah's post and this threw an error when I tried. So kindly ensure that the delete button is not visible to a user who isn't the author of a post or comment so users aren't able to delete other users' posts and comments  :+1: 
- Video Reference

[Screencast from 07-26-2023 01:54:06 PM.webm](https://github.com/stephenkati/blog/assets/102699958/5a08ad64-0908-4967-8de8-abd5a61c5246)

- Project Requirement 
![image](https://github.com/stephenkati/blog/assets/102699958/1e28528d-b9b9-49ee-85b3-c6be1a7d4df1)